compatibility_server/pip_checker.py

```diff
@@ -350,15 +349,14 @@ def _install(self, container: docker.models.containers.Container):
             stderr=True,
             raise_on_failure=False)
         if returncode:
-            environment_error = PIP_ENVIRONMENT_ERROR_PATTERN.search(output)
-            if environment_error:
-                raise PipError(error_msg=environment_error.group('error'),
-                               command=command,
-                               returncode=returncode)
-
-            return PipCheckResult(self._packages,
-                                  PipCheckResultType.INSTALL_ERROR,
-                                  output)
+            pip_install_error = PIP_INSTALL_ERROR_PATTERN.search(output)
+            if pip_install_error:
+                return PipCheckResult(self._packages,
+                                      PipCheckResultType.INSTALL_ERROR,
+                                      output)
+            raise PipError(error_msg=output,
+                           command=command,
+                           returncode=returncode)
```

Comment:
The outer if is checking whether the returncode is 0 or not, and if it is not 0 it will continue to check whether it is related to the pip install itself.